### PR TITLE
ASA - Critical Hours Frontend

### LINKED ISSUE #3898: ASA: Display critical hours for fuel types (front-end)
Once the back-end work for critical hours is complete we need to start displaying the data in the side panel in ASA.

Acceptance Criteria
- [ ] API endpoint that returns fuel type, area and critical hours
- [ ] Display critical hours in table in side panel
- [ ] Edge case: display 'None' if no critical hours available

Additional Info
- the same data will be used in the advisory report (consider storing in Redux)
- Update critical end hour calculation to prevent 'rounding down'

### PR DESCRIPTION
**Pull request by conbrad**
- Implements critical hours frontend hook up to show critical hours in fuel stats table
- A bunch of renaming and refactoring, "fuel types" -> "fuel stats" and the like
- Closes #3898 
# Test Links:
[Landing Page](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3909-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/api/fbaAPI.ts [hunk 1]
```diff
@@ -96,6 +103,12 @@ export interface FuelType {
   description: string
 }
 
+export interface FireCentreHfiFuelsData {
```
Comment (dgboss) [inline web/src/api/fbaAPI.ts:106]: I'm not sure this interface accurately reflects the actual shape of the object? This interface makes it look like a fire centre name will only have one property, it's not clear that there could be multiple properties with type `[fire_zone_id: number]: FireZoneThresholdFuelTypeArea[]`.  I might be old school, but I'd prefer to see an array of `[fire_zone_id: number]: FireZoneThresholdFuelTypeArea[]`. Perhaps @conbrad has a different opinion?

Comment (conbrad) [reply web/src/api/fbaAPI.ts:106]: I agree, I think the endpoint that existed had a lot of rough edges, I can address them in the critical hours PR.

Comment (dgboss) [reply web/src/api/fbaAPI.ts:106]: On second thought, I think this is ok, I'm more used to seeing this syntax in a Type definition. I think it's fine (and a matter of preference).

Comment (conbrad) [reply web/src/api/fbaAPI.ts:106]: Nice catch, I agree, I'll start doing some refactors once the tabs PR is in.

Comment (dgboss) [reply web/src/api/fbaAPI.ts:106]: Oops, I added this comment to the wrong PR.
 Index signatures are totally valid, it's the rigid nature of interfaces from my C# background coming back to haunt me!

File: web/src/features/fba/components/viz/FuelSummary.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ const columns: GridColDef[] = [
   {
     field: 'code',
     headerClassName: 'fuel-summary-header',
-    headerName: 'Fuel Type',
+    headerName: 'Primary Fuels',
```
Comment (brettedw) [inline web/src/features/fba/components/viz/FuelSummary.tsx:39]: Changed this back to 'Fuel Type' after stand up chat this morning

**Comment by brettedw** [comment]
My local db has fuel stats, but no critical hours data (table exists though). I figured it should show me the fuel data with an empty critical hours column but the API returns empty for all stats in each zone so it displays 'No fuel type information available.'
Haven't figured out why that is yet

File: web/src/features/fba/components/viz/CriticalHours.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Typography } from '@mui/material'
```
Comment (dgboss) [inline web/src/features/fba/components/viz/CriticalHours.tsx:1]: Looks like this file is formatted with 4 spaces for indents isntead of 2.

**Comment by conbrad** [comment]
> My local db has fuel stats, but no critical hours data (table exists though). I figured it should show me the fuel data with an empty critical hours column but the API returns empty for all stats in each zone so it displays 'No fuel type information available.' Haven't figured out why that is yet

It was because it was doing an inner join and ignore results that didn't have all fuel stats and critical hours. I adjusted the query, types and tests to perform an outer join so rows with some fuel stats or some critical hours would return values in this commit: https://github.com/bcgov/wps/pull/3909/commits/8484b9bb60ffeb985b4f1deec8cf79d035d10b36

File: web/src/features/fba/components/viz/CriticalHours.tsx (hunk 3 above)
Comment (conbrad) [reply web/src/features/fba/components/viz/CriticalHours.tsx:1]: Reran prettier on this file in: https://github.com/bcgov/wps/pull/3909/commits/6a2c057097a689204cd684a411e253f8226f6e05 

We should get it configured to run on save

**Review by brettedw — APPROVED** [review]
Nice work. Love the endpoint & variable name simplification

File: web/src/features/fba/components/viz/FuelSummary.tsx [hunk 4]
```diff
@@ -57,20 +57,32 @@ const columns: GridColDef[] = [
     renderCell: (params: GridRenderCellParams) => {
       return <FuelDistribution code={params.row['code']} percent={params.row['percent']} />
     }
+  },
+  {
+    field: 'criticalHours',
+    flex: 3,
+    headerClassName: 'fuel-summary-header',
+    headerName: 'Critical Hours',
+    minWidth: 120,
```
Comment (dgboss) [inline web/src/features/fba/components/viz/FuelSummary.tsx:66]: I'm seeing a horizontal scroll bar for the fuel summary table which would be nice to get rid of. I played around a little and we could try:
1. Remove `flex: 3` and `width: 120` and add `minWidth: 80` for the code col def
2. Remove `minWidth: 200` and add `flex: 1` for the area col def
3. Remove `flex: 3` and add `minWidth: 110` for critical hours col def

Comment (conbrad) [reply web/src/features/fba/components/viz/FuelSummary.tsx]: Nice! Yeah works locally for me, added changes here: https://github.com/bcgov/wps/pull/3909/commits/79e4d2937495c714b45050fecf6cbf654a3b68a1

File: api/app/routers/fba.py [hunk 5]
```diff
@@ -161,23 +120,22 @@ async def get_hfi_fuels_data_for_fire_centre(run_type: RunType, for_date: date,
         all_zone_data = {}
         for zone_id in zone_ids:
```
Comment (dgboss) [inline api/app/routers/fba.py:121]: I'm not sure it matters but we're going to be hitting the database with a query for each zone id. I wonder if there would be any performance gain/loss to query by fire center id and get back all the data in one shot? Maybe not worth the effort.

**Review by dgboss — APPROVED** [review]
Looks great!

File: api/app/routers/fba.py (hunk 5 above)
Comment (conbrad) [reply api/app/routers/fba.py:121]: If it's a problem we can address, queries are quite fast the way they are though.

**Pull request merged**